### DISCUSSION TITLE
how to set a fixed size for egui::TextEdit::multiline?

### DISCUSSION THREAD

**Discussion by UnknownSuperficialNight** [emilk/egui#3423] [Q&A] [answered]
Im trying to make a **TextEdit** area that is **scrollable** in my gui but i cant find any thing in the docs for setting a **height** and **width** of  **TextEdit**. I tried to use a **Grid** to try add a **fixed_size** but i think its the wrong tool for the job so the question is. Is there a way to set a **fixed_size** for **TextEdit**?

here is the relevant code:

```
// Create a scrollable area
 egui::ScrollArea::vertical().show(ui, |ui| {
     // add TextEdit widget for file editing
     Grid::new("my_grid")
     .min_col_width(200.0)
     .min_row_height(100.0)
     .show(ui, |ui| {
         egui::TextEdit::multiline(&mut self.downloader_yaml_text)
            .hint_text("Type something!")
            .show(ui);
        });
    });
```

**Comment by Red44** [comment]
I had to face the same issue and found out, that there is **no good solution**. Either you **resize your font** (the height of your TextEdit is determined by its font height) **or you create a new window** that creates a fixed frame. 
Window solution is the one you probably want:
```rust
let fixed_size = vec2(200.0, 100.0);
ui.style_mut().visuals.window_shadow = Shadow::NONE;
ctx.set_style(ui.style().clone());
Window::new("")
    .hscroll(true)
    .movable(false)
    .open(&mut true)
    .current_pos(ui.next_widget_position())
    .fixed_size(fixed_size)
    .resizable(false)
    .title_bar(false)
    .vscroll(true)
    .show(ctx, |ui| {
       ui.add(TextEdit::multiline(&mut "my lines\n".repeat(100)));
    }); 
```
It is really weird, that the normal window allows this TextEdit to go outside the frame and all of that stuff. Functions like max_height etc. do not seem to work either.

**Comment by smarTHh2019** [comment]
For a similar problem I used the following line to insert the sized multiline:
`ui.add_sized(egui::vec2(400.0, 300.0), egui::TextEdit::multiline(&mut text_long));`
maybe it helps you.

**Reply by Xerxes004** [reply]
This works! I solved my issue with:

```rust
egui::ScrollArea::vertical()
    .show(ui, |ui| {
        ui.add_sized(ui.available_size(), myWidget);
    });
```

**Comment by ssyqq** [comment]
Just put it in ScrollArea

**Comment by algorithmiker** [comment, marked as answer]
For setting the width of a TextEdit, `egui::TextEdit::desired_width` seems to work. (so for example for a textedit that takes up just the space of its text, `TextEdit::singleline(text).desired_width(0.0).clip_text(false).show()` works)
This of course affects the minimum, and not the maximum size. For the latter, you can use the `add_sized` method as described here previously.
